valhalla/jawn/src/controllers/datasetController.ts

```diff
@@ -0,0 +1,208 @@
+// src/users/usersController.ts
+import * as Sentry from "@sentry/node";
+import {
+  Body,
+  Controller,
+  Path,
+  Post,
+  Request,
+  Route,
+  Security,
+  Tags,
+} from "tsoa";
+import { postHogClient } from "../lib/clients/postHogClient";
+import { supabaseServer } from "../lib/routers/withAuth";
+import { FilterNode } from "../lib/shared/filters/filterDefs";
+import { getRequests } from "../lib/stores/request/request";
+import { FineTuningManager } from "../managers/FineTuningManager";
+import { JawnAuthenticatedRequest } from "../types/request";
+
+export interface GenerateHashQueryParams {
+  apiKey: string;
+  userId: string;
+  keyName: string;
+}
+
+export async function hasAccessToFineTune(orgId: string) {
+  const { data: org, error: orgError } = await supabaseServer.client
+    .from("organization")
+    .select("*")
+    .eq("id", orgId)
+    .single();
+  if (orgError) {
+    return false;
+  }
+
+  if (!org.tier) {
+    return false;
+  }
+  if (org.tier === "free") {
+    const jobCountQuery = await supabaseServer.client
+      .from("finetune_job")
+      .select("*", { count: "exact" })
+      .eq("organization_id", orgId);
+    console.log("jobCountQuery", jobCountQuery);
```

Comment:
delete

Reply:
ugh not your commented on migrated code  lol . jk it's all good